string blocks
bool gripper_open
